File: backend/internal/data/message.go
package data

import "time"

type Message struct {
	Id             int64     `json:"id"`
	ConversationId int64     `json:"conversationId"`
	Content        string    `json:"content"`
	SenderId       int64     `json:"senderId"`
	PrevMessageId  int64     `json:"prevMessageId"`
	CreatedAt      time.Time `json:"createdAt"`
}

type PostMessageDto struct {
	ConversationId int64  `json:"conversationId"`
	Content        string `json:"content"`
	PrevMessageId  int64  `json:"prevMessageId"`
}

File: frontend/src/app/_models/message.ts
import { User } from './user';

export interface Message {
  id: number;
  senderId: number;
  conversationId: number;
  prevMessageId: number;
  createdAt: Date;
  content: string;
}

export interface MessageDto {
  conversationId: number;
  prevMessageId: number;
  content: string;
}
